stdlib/pervasives.gr

```diff
@@ -218,6 +219,19 @@ export (<=)
  */
 export (>=)
 
+/**
+ * Compares the first argument to the second argument and produces an integer result.
+ * Provides a consistent ordering over all types and is suitable for sorting and other kinds of ordering.
+ * `compare` treats `NaN` differently than the other comparison operators in that it considers `NaN` equal to itself and smaller than any other number.
+ *
+ * @param num1: The first operand
+ * @param num2: The second operand
+ * @returns A negative integer if the first operand is less than the second operand, `0` if they are equal, or a positive integer otherwise
```

Comment:
It's not just `-1` and `1`? Is there some sort of scale?

Reply:
There's no scale! I believe languages implement it this way just to make it exhaustive and allow simple implementations like `a - b` that don't need any branching logic. I believe this implementation will ever only return one of `-1, 0, 1` though (I'd need to double-check a few things).